browser/playlists/playlists_service.cc

```diff
@@ -0,0 +1,88 @@
+/* Copyright (c) 2019 The Brave Authors. All rights reserved.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+#include "brave/browser/playlists/playlists_service.h"
+
+#include <string>
+#include <utility>
+
+#include "base/files/file_path.h"
+#include "base/files/file_util.h"
+#include "base/task/post_task.h"
+#include "base/task_runner_util.h"
+#include "brave/components/playlists/browser/playlists_controller.h"
+#include "content/public/browser/browser_context.h"
+
+#if !defined(OS_ANDROID)
+#include "brave/browser/playlists/desktop_playlists_player.h"
+#include "chrome/browser/profiles/profile.h"
+#endif
+
+namespace brave_playlists {
+namespace {
+
+const base::FilePath::StringType kBaseDirName(FILE_PATH_LITERAL("playlists"));
+
+}  // namespace
+
+PlaylistsService::PlaylistsService(content::BrowserContext* context)
+    : base_dir_(context->GetPath().Append(kBaseDirName)),
+#if !defined(OS_ANDROID)
```

Comment:
why is this in an OS_ANDROID guard?

Reply:
also it seems like you have a mix of OS_ANDROID and ENABLE_EXTENSIONS guards, but it seems like they should all be `ENABLE_EXTENSIONS`

Reply:
you should only use OS_ANDROID guards for thing that are android-specific, not feature-specific (extensions).